server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -1822,8 +1822,8 @@ public void testForceMergeWithSoftDeletesRetentionAndRecoverySource() throws Exc
             )
         ) {
             int numDocs = scaledRandomIntBetween(10, 100);
+            boolean useRecoverySource = randomBoolean() || omitSourceAllTheTime;
```

Comment:
Move the `useRecoverySource` flag out of the for loop makes every document has same flag, the randomness is lost, I think we need to keep the randomness in order to cover different cases.